### ISSUE TITLE
[Request] Allow for updating tags in m3u playlist from foobar2000

### ISSUE THREAD

**Issue by VideogameScrapbook** [losnoco/vgmstream#566]
The m-TAGS component has a really cool feature where you can easily create a !.tags file from the player itself, then edit the tags using the standard foobar2000 tagging functions (including bulk editing), where anything you edit is saved to the !.tags file:
http://www.m-tags.org/foobar2000_component.html

The special m3u format that vgmstream supports (described here) does not support editing tags from foobar2000 directly:
https://github.com/losnoco/vgmstream#Tagging

Could you update vgmstream to support adding tags from foobar2000 to !tags.m3u files? If the effort is too large, maybe it could be mentioned in the readme that m-TAGS is an alternative format for tagging that offers that functionality, to increase awareness? My intention is to reduce the barrier of entry for creating tags for streaming files so that more people could contribute to the effort.

**Comment by bnnm** [comment]
I'll have to test it, but the main issue against those features is time. There'd be a lot to learn about foobar's API to make something like that, and it would only help foobar users (I aim to make features mostly the same for all players/plugins). I'll add a mention about m-TAGS in the readme though.

**Comment by VideogameScrapbook** [comment]
Makes sense, thank you for offering to update readme. Now to find somebody who has the knowhow and willingness to host some sort of repository for M3U and/or m-TAGS files for stream music. Would be great to make it easier for a community effort to tag albums without a lot of re-uploading/re-downloading of ZIPs, to be able to know what song is what without trial and error or listening to MP3/FLAC tagged conversions.

**Comment by bnnm** [comment]
Note that there seems to be another, more recent plugin for external tags: https://foobar.hyv.fi/?view=foo_external_tags. Not sure about its advantages though.

For a tag repository you could just allow any format instead of having to choose, it shouldn't be too hard to convert between them with some program. That way people can use whatever they are confortable with.

**Comment by VideogameScrapbook** [comment]
Before I settled on using m-TAGS, I believe I looked at that format as well and decided m-TAGS worked better for whatever reason. If I looked into switching the format I use, then I'd probably switch to your m3u format now that I'm aware of it and know how to do workarounds (especially if mp3tag can be used for creating your m3u format).

I agree the tag repository could support any textual format that contributors wanted to give. I bet a format would be more common than others, and I'd be fine with it being the one you developed especially if mp3tag can be used. :)
